examples/hosts/app-integration/schema-upgrade/src/migrator/migrator.ts

```diff
@@ -172,11 +153,39 @@ export class Migrator extends TypedEventEmitter<IMigratorEvents> implements IMig
 
             this._migrationP = undefined;
 
+            this.taskManager.off("lost", exitMigration);
+            this.taskManager.off("completed", exitMigration);
+            this.taskManager.off("assigned", onAssigned);
+            this.taskManager.complete(this.migrationTaskId);
+
+            this.takeAppropriateActionForCurrentMigratable();
+        };
+
+        const exitMigration = (taskId: string) => {
+            if (taskId !== this.migrationTaskId) {
+                return;
+            }
+            this.taskManager.off("lost", exitMigration);
+            this.taskManager.off("completed", exitMigration);
+            this.taskManager.off("assigned", onAssigned);
+            migratedModel?.close();
+            this._migrationP = undefined;
             this.takeAppropriateActionForCurrentMigratable();
         };
 
+        const onAssigned = (taskId: string) => {
+            if (taskId !== this.migrationTaskId) {
+                return;
+            }
+            this._migrationP = doTheMigration().catch(console.error);
```

Comment:
```suggestion
            this._migrationP = doTheMigration().catch((error) => {
                // Allow next client to retry if we fail
                console.error(error);
                this.taskManager.abandon(this.migrationTaskId);
            });
```

@ChumpChief, do you think this would be sufficient to handle retry scenarios? Basically, if we error out of the migration then we would give up the task and let the next client in queue try. We would also automatically re-enter the queue because we subscribed to the task. Let me know what you think!

Reply:
I put a couple notes above about reusing attached containers and such - although looking to other clients to pick up the task may work OK, probably consider a retry path for the local client to minimize extra containers getting attached and to be a little faster.  This is most true in the scenario that a client is the only one connected to the collaboration session - abandoning and re-volunteering introduces a delay that we can probably avoid.